src/System.Private.CoreLib/shared/Internal/Win32/RegistryKey.cs

```diff
@@ -162,7 +163,7 @@ public unsafe string[] GetValueNames()
             // add up quickly- we'll try to keep the memory pressure low and grow the buffer
             // only if needed.
 
-            char[] name = ArrayPool<char>.Shared.Rent(100);
+            char[]? name = ArrayPool<char>.Shared.Rent(100);
```

Comment:
Why is this nullable?

Reply:
we are resizing this array and resize functions takes a nullable array as argument

Reply:
But Array.Resize will never set it to null.  The right fix for now is to use `!` on the array argument to Resize, with a TODO-NULLABLE comment for https://github.com/dotnet/roslyn/issues/26761.